R-package/R/lightgbm.R

```diff
@@ -92,6 +92,18 @@ NULL
 #' @inheritParams lgb_shared_params
 #' @param label Vector of labels, used if \code{data} is not an \code{\link{lgb.Dataset}}
 #' @param weight vector of response values. If not NULL, will set to dataset
+#' @param nthreads Number of parallel threads to use. For best speed, this should be set to the number of
+#'                 physical cores in the CPU - in a typical x86-64 machine, this corresponds to half the
+#'                 number of maximum threads (e.g. `nthreads = max(parallel::detectCores() / 2L, 1L)` as
+#'                 a shorthand for the optimal value).
+#'
+#'                 Be aware that using too many threads can result in speed degradation in smaller datasets
+#'                 (see the parameters documentation for more details).
+#'
+#'                 If passing zero, will use the default number of threads configured for OpenMP
+#'                 (typically controlled through an environment variable `OMP_NUM_THREADS`).
+#'
+#'                 This parameter overrides `num_threads` in `params` if it exists there.
```

Comment:
```suggestion
```

Across LightGBM's R and Python packages, we try to enforce the behavior that elements in `params` (a list in R and a dictionary in Python) always take precedence over keyword arguments to individual functions.

Please remove this line in the documentation.